File: platform/src/java/com/xcase/rest/generator/RESTServiceDefinition.java
package com.xcase.rest.generator;

import com.xcase.rest.generator.IServiceDefinition;
import java.lang.invoke.MethodHandles;
import java.net.URI;
import java.util.*;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class RESTServiceDefinition implements IServiceDefinition
{
    public List<ClassDefinition> Classes;
    public String EndPoint;
    public String ObjectsPackageName;
    public String[] SourceStrings;
    public HashMap<String, ProxyDefinition> ProxyClasses = new HashMap<String, ProxyDefinition>();
    
    public RESTServiceDefinition() {
        
    }
    
    public RESTServiceDefinition(String objectsPackageName) {
        this.ObjectsPackageName = objectsPackageName;
    }    

    public String getEndPoint() {
        return EndPoint;
    }

    public String[] getSourceStrings() {
        return SourceStrings;
    }

    public HashMap<String, ProxyDefinition> getProxyClasses() {
        return ProxyClasses;
    }
    
    public void writeToStringBuilders() {
        for (String proxy : ProxyClasses.keySet()) {
            ProxyDefinition proxyDefinition = ProxyClasses.get(proxy);
            proxyDefinition.writeToStringBuilder(proxy, ObjectsPackageName, "");
        }
        
        for (ClassDefinition classDefinition : Classes) {
            classDefinition.writeToStringBuilder(ObjectsPackageName);
        }        
    }
}

File: dotnetcore/XCaseServiceClient/XCase.Swagger.ProxyGenerator/RESTServiceDefinition.cs
namespace XCase.REST.ProxyGenerator
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;
    using System.Threading.Tasks;
    using XCase.ProxyGenerator;

    public class RESTServiceDefinition : IServiceDefinition
    {
        public string EndPoint { get; set; }
        public string[] SourceStrings { get; set; }
        public List<string> ProxyClasses = new List<string>();

        public string GetEndPoint()
        {
            return EndPoint;
        }

        public string[] GetSourceStrings()
        {
            return SourceStrings;
        }

        public List<string> GetProxyClasses()
        {
            return ProxyClasses;
        }

        public void SetEndPoint(string endPoint)
        {
            EndPoint = endPoint;
        }

        public void SetProxyClasses(List<string> proxyClasses)
        {
            ProxyClasses = proxyClasses;
        }

        public void SetSourceStrings(string[] sourceStrings)
        {
            SourceStrings = sourceStrings;
        }
    }
}
